### ISSUE TITLE
Weird simulation results due to checkpoints.

### ISSUE THREAD

**Issue by finnolec** [ComputationalRadiationPhysics/picongpu#3899]
Hello, I have found some issues with the checkpoints with the current `dev` branch (54efd251bc0df62a8f7e84b98ca5fd7f8e0c7718).  For this run I used the `LaserWakefield` example, I increased all `static constexpr` in `DefaultExchangeMemCfg` in the `memory.param` by a factor of 8 (to clean up the `stderr`, the simulation output looks identical even if the memory.param is not changed). 
I ran the first simulation on the `k80` partition on hemera where I changed the following lines in `4.cfg`:
```shell
TBG_devices_y=8

TBG_steps="4096"

TBG_checkpoints="--checkpoint.period 2048:2048 --checkpoint.backend openPMD"
TBG_plugins="!TBG_pngYX                    \
             !TBG_e_histogram              \
             !TBG_e_PSypy                  \
             !TBG_e_macroCount             \
             !TBG_checkpoints"

```
With my second simulation I used the checkpoint at `2048` by changing the checkpoints command from above to
```shell
TBG_checkpoints="--checkpoint.restart --checkpoint.backend openPMD"
```
and I created the output files with `tbg` without the `-s` flag. Then I went to the directory of the second simulation, created a `simOutput` folder and created a symlink to the checkpoints from the first simulation `ln -s PATH/TO/CHECKPOINT/FOLDER .`

What happens now is, that the png from the first simulation from before the checkpoint (2000) looks like this:
![e_png_yx_0 5_002000](https://user-images.githubusercontent.com/45068748/140287693-9a62d99e-6b1a-440b-ade3-306458df16e6.png)
And after the checkpoint (2100) (in the first simulation) it looks like this:
![e_png_yx_0 5_002100](https://user-images.githubusercontent.com/45068748/140287766-2fa762bf-1f9b-4ca1-ac05-98c92e3d9c84.png)
But in the second simulation, with the checkpoint-restart, the same timestep (2100) looks like this:
![e_png_yx_0 5_002100](https://user-images.githubusercontent.com/45068748/140287925-51d84f89-bfe0-4b17-8fd0-4c96ff769d8a.png)
So there is something wrong in the restart... Since we can see the 7 transversal/vertical lines I would assume that it comes from me using 8 longitudinal gpus, but I didn't check that. And the bubbles look obviously very different too.

@steindev told me to tag @sbastrakov and @Anton-Le

**Comment by sbastrakov** [comment]
Good catch. Can confirm the bug exists on the current `dev`, reproduces for me every time with moving window and restarting. Investigating.

**Comment by sbastrakov** [comment]
Currently it seems either writing or reading is broken, not the calculation part.

**Comment by psychocoderHPC** [comment]
solved with #3902
